contracts/lease/src/contract/endpoins.rs

```diff
@@ -125,10 +125,18 @@ pub fn sudo(deps: DepsMut<'_>, env: Env, msg: SudoMsg) -> ContractResult<CwRespo
 }
 
 #[cfg_attr(feature = "contract-with-bindings", entry_point)]
-pub fn query(deps: Deps<'_>, env: Env, _msg: StateQuery) -> ContractResult<Binary> {
+pub fn query(deps: Deps<'_>, env: Env, msg: QueryMsg) -> ContractResult<Binary> {
     state::load(deps.storage)
-        .and_then(|state| state.state(env.block.time, &deps.querier))
-        .and_then(|resp| to_binary(&resp).map_err(Into::into))
+        .and_then(|state| match msg {
+            QueryMsg::State {} => state
+                .state(env.block.time, &deps.querier)
+                .and_then(|resp| to_binary(&resp).map_err(Into::into)),
+            QueryMsg::IsClosed {} => to_binary(&matches!(
+                state,
+                State::Closed { .. } | State::Liquidated { .. },
+            ))
```

Comment:
The knowledge of which lease states are considered final should be brought into the states themselves. Otherwise, if we add another state we must keep in mind updating this list too. So, please introduce the corresponding trait method with **no default implementation**.